src/plugin/timezone/index.js

```diff
@@ -128,10 +133,13 @@ export default (o, c, d) => {
   d.tz = function (input, arg1, arg2) {
     const parseFormat = arg2 && arg1
     const timezone = arg2 || arg1 || defaultTimezone
-    const previousOffset = tzOffset(+d(), timezone)
-    if (typeof input !== 'string') {
-      // timestamp number || js Date || Day.js
-      return d(input).tz(timezone)
+    const previousOffset = tzOffset(+d(input, parseFormat), timezone)
+    // To differentiate date only string (e.g. yy-mm-dd) from date string with negative
+    // 2-digit offset (hour offset zz, e.g. yy-mm-zz) we require at least 8 characters
+    // before offset (i.e. yy-mm-dd-zz)
+    if ((typeof input !== 'string') || /.{8,}[+-]\d\d:?(\d\d)?$|Z$/i.test(input)) {
```

Comment:
Found another issue:
If we parse date in format YYYY-MMM-DD (2023-Sep-03) it will pass this check and return a wrong date.  
8 characters check should be removed from regex expression and` /[+-]\d\d:\d\d$|Z$/i` should be good enough check if we have a string with offset

Reply:
But then what about 2-digit only offsets (which are completely permitted (see the documentation for [moment](https://momentjs.com/docs/#/parsing/string/))?
A time offset can be `+01:30`, but also `-02` (describing a -2 hours offset).

So what does `10-03-04` mean? `10-03` with a 4-hour offset or `10-03-04` with no offset?

Reply:
Oh I understand. Sorry I was not aware of that as 2 digit only offset was not used in our case.